src/BitIntegers.jl

```diff
@@ -311,14 +311,40 @@ promote_rule(::Type{Float16}, ::Type{<:XBI}) = Float16
 (|)(x::T, y::T) where {T<:XBI} = or_int(x, y)
 xor(x::T, y::T) where {T<:XBI} = xor_int(x, y)
 
->>( x::UBS, y::UBU) = ashr_int(x, y)
->>( x::UBU, y::UBU) = lshr_int(x, y)
->>>(x::UBI, y::UBU) = lshr_int(x, y)
-<<( x::UBI, y::UBU) = shl_int(x, y)
-
->>( x::UBI, y::Int) = ifelse(0 <= y, x >> unsigned(y),  x << unsigned(-y))
-<<( x::UBI, y::Int) = ifelse(0 <= y, x << unsigned(y),  x >> unsigned(-y))
->>>(x::UBI, y::Int) = ifelse(0 <= y, x >>> unsigned(y), x << unsigned(-y))
+# LLVM bit shifting (lshr_int, ashr_int and shl_int) has performance issues for large integers
+# The following functions allows much faster code (10x speedup for 1024 bit integers)
+
+# Note: this should be a power of 2
+const SHIFT_SPLIT_NBITS = 64
+
+@inline @generated function shift_call(sh_fun::Function, x::I, y::UBU) where {I<:XBI}
+    nbits = 8 * sizeof(I)
+    # Performance issue does not afect integers with less than 256 bits
```

Comment:
```suggestion
    # performance issue does not affect integers with no more than 128 bits
```
(we can have e.g. `Int200` in between)

Reply:
I actually didn't test the performance for integers between `Int128` and `Int256`, so I'm not sure if `Int200` would have performance issue.

Reply:
Ok, so I think my suggestion makes sense then? As the fast-path you write is for `nbits <= 128`.